From: Michael Chan <mchan@broadcom.com>
Date: Wed, 4 Jan 2012 12:12:27 +0000
Subject: [PATCH 1/5] cnic: Re-init dev->stats_addr after chip reset
Patch-mainline: v3.3-rc1
Git-commit: a9e0a4f2ca5e97ae2cff0bda72b9645e047c1a3d
References: bnc#790592 FATE#313914

because bnx2x frees the old and allocates new memory during chip reset.

Signed-off-by: Michael Chan <mchan@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/broadcom/cnic.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/ethernet/broadcom/cnic.c
+++ b/drivers/net/ethernet/broadcom/cnic.c
@@ -4869,6 +4869,7 @@ static int cnic_start_bnx2x_hw(struct cn
 	int func = CNIC_FUNC(cp), ret;
 	u32 pfid;
 
+	dev->stats_addr = ethdev->addr_drv_info_to_mcp;
 	cp->port_mode = CHIP_PORT_MODE_NONE;
 
 	if (BNX2X_CHIP_IS_E2_PLUS(cp->chip_id)) {
